### ISSUE TITLE
Lane and voice assist is skipped when "avoid toll roads" is set

### ISSUE THREAD

**Issue by mocsa** [osmandapp/OsmAnd#12438]
How to reproduce:
The M0 trunk highway at 47.57N, 19.1631E becomes a toll road. When you drive from South to North, this point is your last chance to exit to avoid paying a toll.

If you do a simulated Navigation from 47.55727N, 19.17614E to 47.58288N, 19.15123E and "avoid toll roads" is switched OFF in your navigation settings, then you will see corrent lane assist at every junction. Also, OsmAnd will tell you to "turn slightly right" before the exit, therefore warning you not to enter the toll section.

However, if you navigate while "avoid toll roads" is switched ON, then the announcement and the lane assist is skipped for the first junction and immediately the second junction is shown where you have to keep left. It is not easy to recognize that the assist is for the second junction, because at this place there are several junctions very close to each other. If you keep left at the first junction then you will actually ENTER the toll road. This is very confusing and dangerous! I actually had to pay a fine because of this.

To summarize, the navigation setting "avoid toll roads" changes the voice announcements and lane assist display. It seems the problem happens at the node where the toll and non-toll roads separate. Maybe OsmAnd thinks that since I selected "avoid toll roads", I will "automatically" exit where the toll road starts, without needing any assistance? It is just the opposite! I want OsmAnd to shout at me to exit so I can avoid the toll.

I'm using OsmAnd+ 3.9.10.

**Comment by vshcherb** [comment]
Full turn information at http://www.openstreetmap.org/#map=19/47.57024/19.163246 is completely skipped - turn = "Turn slightly right(C|C|+TSLR).

**Comment by mocsa** [comment]
Shall I (can I?) do something to fix this problem. It is not clear from your answer.

Also, while we are at it, I saw these kind of debug labels (C|C|+TSLR) in OsmAndMapCreator when I tested the Routing, although they are close to unreadable when multiple labels are on top of each other. What do they mean? Are they documented somewhere?

**Comment by vshcherb** [comment]
I confirmed the issue and added some details, we need to fix it somehow hopefully this release
